WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleCardPresentPaymentInLineMessage.swift

```diff
@@ -16,17 +16,17 @@ struct PointOfSaleCardPresentPaymentInLineMessage: View {
         case .tapSwipeOrInsertCard(let viewModel):
             PointOfSaleCardPresentPaymentTapSwipeInsertCardMessageView(viewModel: viewModel)
         case .processing:
-            Text("processing...")
+            POSCardPresentPaymentMessageView(viewModel: POSCardPresentPaymentMessageViewModel(title: "processing..."))
```

Comment:
💭 for the 3 `POSCardPresentPaymentMessageView` use cases in this file, how about creating a separate view for each case so that they can be customized more easily? For example, if we want to add some animation only in the success case. If we really want to reuse some SwiftUI components like the title, they can still be shared within their own views.

I also felt like it's a bit odd to reuse `POSCardPresentPaymentMessageView` with other use cases in the PR to show buttons and progress view. Having separate views for different purposes and reusing only what's common is what I tend to follow using SwiftUI.

Reply:
There were/are special views for a lot of cases but since they show the content in the same layout/design I was thinking of having one view to show that data so we do not have 10 places to update if we change a color or a font.
I was even thinking of removing `PointOfSaleCardPresentPaymentTapSwipeInsertCardMessageView` and similar views and just use `POSCardPresentPaymentMessageView`.

There is no rush with this PR so I will maybe wait when I get those animated icons and see what would be the best approach and how many view structs to have.

Reply:
I agree that there are UI elements that can be reused like the title and message for states that share the same layout. For `POSCardPresentPaymentMessageView`, could we try moving the optional image, circle, progress view, and buttons out to separate views, and replace the image/circle/progress part with a generic `topContent: View` and button container with a generic `bottomContent: View`? This way, if we touch the layout of the optional image/circle/progress/buttons for a subset of states, we don't have to worry about affecting the UI of other states.

Maybe we can have a third/fourth opinion from @iamgabrielma and @staskus as well! I think this affects how we develop UI down the road and is worth discussing as a team.

Reply:
> There is no rush with this PR so I will maybe wait when I get those animated icons and see what would be the best approach and how many view structs to have.

I agree with the iterative process, right now for me it's difficult to say what an ideal setup for these views would look like, depending on how many similarities and differences we'll have. Choosing to actively refactor while developing could help us find the balance between customization and reusability. 

> I was thinking of having one view to show that data so we do not have 10 places to update if we change a color or a font.

Yes, agree that having a completely different view for every different configuration won't make sense if we want to share some if not most attributes between different views.

> ..could we try moving the optional image, circle, progress view, and buttons out to separate views <..> This way, if we touch the layout of the optional image/circle/progress/buttons for a subset of states, we don't have to worry about affecting the UI of other states.

Something like that seems reasonable to me. The main view could share certain styling attributes with child views so we wouldn't need to have 10 places to update if we change a color or a font. Having 4 optional members and one non-optional generates lots of possible permutations of this view. I imagine the number of members will expand as views become more and more complex. We would notice the complexity if we tried to create previews that would cover all the possible scenarios that this UI should theoretically handle.

Reply:
> There is no rush with this PR so I will maybe wait when I get those animated icons and see what would be the best approach and how many view structs to have.

💯 

> Having 4 optional members and one non-optional generates lots of possible permutations of this view. I imagine the number of members will expand as views become more and more complex

Agree with this, I've also dropped a similar [comment](https://github.com/woocommerce/woocommerce-ios/pull/13166#pullrequestreview-2152591848) for the viewmodel where we might be trying to handle too many cases for the sake of reusability. Jaclyn's suggestion of having some main view with only the reusable components, and move the optional ones to sub-views feels a good approach to explore, this could even be managed internally by the view model we initialize the subviews with, so it wouldn't need to be exposed.